File: app/src/main/java/com/zzzmode/appopsx/ui/model/AppPermissions.java
package com.zzzmode.appopsx.ui.model;

import java.util.List;

public class AppPermissions {

  public AppInfo appInfo;
  public List<OpEntryInfo> opEntries;

  public boolean hasPermissions() {
    return opEntries != null && !opEntries.isEmpty();
  }

  @Override
  public String toString() {
    return "AppPermissions{" +
        "appInfo=" + appInfo +
        ", opEntries=" + opEntries +
        '}';
  }
}

File: app/src/main/kotlin/com/zzzmode/appopsx/ui/model/AppPermissions.kt
package com.zzzmode.appopsx.ui.model

class AppPermissions(val appInfo: AppInfo) {

    var opEntries: List<OpEntryInfo>? = null

    fun hasPermissions(): Boolean {
        return opEntries != null && !opEntries!!.isEmpty()
    }

    override fun toString(): String {
        return "AppPermissions{" +
                "appInfo=" + appInfo +
                ", opEntries=" + opEntries +
                '}'.toString()
    }
}